File: sprint4/By languages/Java/lesson1code4.java
import java.util.ArrayList;

class Pair<K, V> {
  public K key;
  public V value;

  public Pair(K key, V value) {
    this.key = key;
    this.value = value;
  }
}

class Map<K, V> {
  private ArrayList<Pair<K, V>> pairs = new ArrayList<Pair<K, V>>();

  public V get(K key) {
    for (Pair<K, V> pair : pairs) {
      if (pair.key.equals(key)) { // пара с указанным ключом найдена
        return pair.value;
      }
    }
    return null;
  }

  public void set(K key, V value) {
    for (Pair<K, V> pair : pairs) {
      if (pair.key.equals(key)) { // пара с указанным ключом найдена
        // обновить значение в найденной паре
        pair.value = value;
        return;
      }
    }
    // пара с заданным ключом не найдена
    // добавить пару (key, value) в pairs
    pairs.add(new Pair<K, V>(key, value));
  }
}

File: sprint4/By languages/C++/lesson1code4.cpp
#include <vector>
#include <utility>

struct Pair {
    std::string key;
    std::string value;
};

class Map {
private:
    std::vector<Pair> pairs;
public:
    std::string get(std::string key) {
        for (auto pair : pairs) {
            if (pair.key == key) { // пара с указанным ключом найдена
                return pair.value;
            }
        }
        return ""; // пара с указанным ключом не найдена
    }

    void set(std::string key, std::string value) {
        for (auto& pair : pairs) {
            if (pair.key == key) { // пара с указанным ключом найдена
                // обновить значение в найденной паре
                pair.value = value;
                return;
            }
        }
        // пара с заданным ключом не найдена
        pairs.emplace_back(Pair{key, value});
    }
};
